### PR TITLE
Add unit tests for DynamoUtils.java

### PR DESCRIPTION
**Pull request by sudo-jaa**
Adding unit tests for DynamoUtil.java

Waiting on Mel to be back in next week to get information on how to deal with license headers and such for this one. 

More tests for this class exist but use Mocking/member matching and i'm going to avoid pulling in any extra dependencies for this project - it's all done via a remote pom so I don't want to interfere too much. 

`By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.`

### CODE REVIEW

File: amazon-kinesis-client/src/test/java/software/amazon/kinesis/leases/DynamoUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,412 @@
...
+    final HashMap<String, AttributeValue> dynamoRecord = new HashMap<String, AttributeValue>();
+    final String key = "?";
+
+    // Act
+    final Long retval = DynamoUtils.safeGetLong(dynamoRecord, key);
+
+    // Assert result
+    Assert.assertNull(retval);
+  }
+}
```
Comment (paulkennethkent) [inline amazon-kinesis-client/src/test/java/software/amazon/kinesis/leases/DynamoUtilsTest.java]: newline

**Review by JohnLBergqvist — CHANGES_REQUESTED** [review]
A lot of redundant tests, given you have multiple tests for the same functions that assert `null`. Would be best to trim these down I feel.

**Review by petersear — COMMENTED** [review]
Agree with John's comment, otherwise looks good.

File: amazon-kinesis-client/src/test/java/software/amazon/kinesis/leases/DynamoUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+    Assert.assertNull(retval);
+  }
+
+  @Test
+  public void safeGetStringInput1NotNullOutputNull5() {
+
+    // Arrange
+    final HashMap<String, AttributeValue> dynamoRecord = new HashMap<String, AttributeValue>();
+    dynamoRecord.put("\"& %", null);
+    final String key = "!E\"%";
```
Comment (JohnLBergqvist) [inline amazon-kinesis-client/src/test/java/software/amazon/kinesis/leases/DynamoUtilsTest.java]: This test is unnecessary - duplicates the one above essentially

**Review by JohnLBergqvist — CHANGES_REQUESTED** [review]

File: amazon-kinesis-client/src/test/java/software/amazon/kinesis/leases/DynamoUtilsTest.java [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+    // Act
+    final List<String> retval = DynamoUtils.safeGetSS(dynamoRecord, key);
+
+    // Assert result
+    final ArrayList<String> arrayList = new ArrayList<String>();
+    Assert.assertEquals(arrayList, retval);
+  }
+
+  @Test
+  public void safeGetSSInput1NotNullOutput2() {
```
Comment (petersear) [inline amazon-kinesis-client/src/test/java/software/amazon/kinesis/leases/DynamoUtilsTest.java]: Similar to safeGetSSInput1NotNullOutput()

File: amazon-kinesis-client/src/test/java/software/amazon/kinesis/leases/DynamoUtilsTest.java [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+    // Act
+    final List<String> retval = DynamoUtils.safeGetSS(dynamoRecord, key);
+
+    // Assert result
+    final ArrayList<String> arrayList = new ArrayList<String>();
+    Assert.assertEquals(arrayList, retval);
+  }
+
+  @Test
+  public void safeGetSSInput1NotNullOutput3() {
```
Comment (petersear) [inline amazon-kinesis-client/src/test/java/software/amazon/kinesis/leases/DynamoUtilsTest.java]: Similar to safeGetSSInput1NotNullOutput()

File: amazon-kinesis-client/src/test/java/software/amazon/kinesis/leases/DynamoUtilsTest.java [hunk 5]
```diff
@@ -0,0 +1,337 @@
...
+    // Act
+    final List<String> retval = DynamoUtils.safeGetSS(dynamoRecord, key);
+
+    // Assert result
+    final ArrayList<String> arrayList = new ArrayList<String>();
+    Assert.assertEquals(arrayList, retval);
+  }
+
+  @Test
+  public void safeGetSSInput1NotNullOutput4() {
```
Comment (petersear) [inline amazon-kinesis-client/src/test/java/software/amazon/kinesis/leases/DynamoUtilsTest.java]: Similar to safeGetSSInput1NotNullOutput()

File: amazon-kinesis-client/src/test/java/software/amazon/kinesis/leases/DynamoUtilsTest.java [hunk 6]
```diff
@@ -0,0 +1,337 @@
...
+    // Act
+    final List<String> retval = DynamoUtils.safeGetSS(dynamoRecord, key);
+
+    // Assert result
+    final ArrayList<String> arrayList = new ArrayList<String>();
+    Assert.assertEquals(arrayList, retval);
+  }
+
+  @Test
+  public void safeGetSSInput1NotNullOutput5() {
```
Comment (petersear) [inline amazon-kinesis-client/src/test/java/software/amazon/kinesis/leases/DynamoUtilsTest.java]: Similar to safeGetSSInput1NotNullOutput()

File: amazon-kinesis-client/src/test/java/software/amazon/kinesis/leases/DynamoUtilsTest.java [hunk 7]
```diff
@@ -0,0 +1,337 @@
...
+    // Assert result
+    Assert.assertNull(retval);
+  }
+
+  @Test
+  public void safeGetLongInput1NotNullOutputNull5() {
+
+    // Arrange
+    final HashMap<String, AttributeValue> dynamoRecord = new HashMap<String, AttributeValue>();
+    dynamoRecord.put("aAAAA", null);
```
Comment (petersear) [inline amazon-kinesis-client/src/test/java/software/amazon/kinesis/leases/DynamoUtilsTest.java]: Similar to safeGetLongInput1NotNullOutputNull()

File: amazon-kinesis-client/src/test/java/software/amazon/kinesis/leases/DynamoUtilsTest.java [hunk 8]
```diff
@@ -0,0 +1,337 @@
...
+
+    // Act
+    final Long retval = DynamoUtils.safeGetLong(dynamoRecord, key);
+
+    // Assert result
+    Assert.assertNull(retval);
+  }
+
+  @Test
+  public void safeGetLongInput1NotNullOutputNull3() {
```
Comment (petersear) [inline amazon-kinesis-client/src/test/java/software/amazon/kinesis/leases/DynamoUtilsTest.java]: Similar to safeGetLongInput1NotNullOutputNull()

File: amazon-kinesis-client/src/test/java/software/amazon/kinesis/leases/DynamoUtilsTest.java [hunk 9]
```diff
@@ -0,0 +1,337 @@
...
+
+    // Act
+    final Long retval = DynamoUtils.safeGetLong(dynamoRecord, key);
+
+    // Assert result
+    Assert.assertNull(retval);
+  }
+
+  @Test
+  public void safeGetLongInput1NotNullOutputNull6() {
```
Comment (petersear) [inline amazon-kinesis-client/src/test/java/software/amazon/kinesis/leases/DynamoUtilsTest.java]: Similar to safeGetLongInput1NotNullOutputNull2()

**Review by petersear — CHANGES_REQUESTED** [review]
I've marked some tests as being similar to others ie. same inputs and assert for the same function.

**Review by petersear — APPROVED** [review]

**Review by JohnLBergqvist — APPROVED** [review]

**Pull request closed without merging**